### DISCUSSION TITLE
Best way to set up with nginx on the same server?

### DISCUSSION THREAD

**Discussion by daninus14** [fukamachi/clack#194] [Q&A]
What's the best way to set up a clack server to work with nginx?

There seem to be two ways in their tutorial:
- with FastCGI https://nginx.org/en/docs/beginners_guide.html#fastcgi
- A regular reverse proxy https://nginx.org/en/docs/beginners_guide.html#proxy

Is it possible to set up clack as FastCGI? Is it recommended? What's the difference between the two?

**Comment by fukamachi** [comment]
I personally use Nginx as a reverse proxy forwarding to Woo.
Clack used to have a FastCGI handler, but it's deleted since cl-fastcgi isn't available on Quicklisp anymore.
https://github.com/fukamachi/clack/pull/189#issuecomment-2264380794

**Reply by daninus14** [reply]
Thanks for answering. That's how I was setting it up. Would it not be faster with woo running it as cgi instead of http, or is woo only http?

**Reply by fukamachi** [reply]
Woo is only for HTTP but should be faster with a UNIX domain socket. I never taken a benchmark with cl-fastcgi.

**Reply by daninus14** [reply]
Interesting: 
- while cl-fastcgi was taken out of quicklisp, sb-fastcgi is still there https://www.quicklisp.org/beta/releases.html 
- and [cl-fastcgi](https://github.com/KDr2/cl-fastcgi?tab=readme-ov-file) is based on sb-fastcgi (they are by the same guy) so why would they take out one but not the other?

**Reply by fukamachi** [reply]
Interesting.
Although I can't find any reasons for cl-fastcgi to be deleted, Zach usually removes libraries when they don't work on the relatively new SBCL and the author doesn't respond for a certain period.
If you find it works on the latest SBCL, there's probably a chance to make it available again.

**Reply by daninus14** [reply]
Ok, thanks, I'll look into it when I have a chance. I'm using woo anyways, so it may not make that much of a difference since only Hunchentoot supports FastCGI. Do you know how hard it would be to add FastCGI to woo? Is it worth it for me to look into it on how to implement it? Since Hunchentoot works with FastCGI I imagine that We/I could probably leverage most of that code to give it support, but I've never dealt with any of this before so I don't know

**Reply by vindarel** [reply]
In my humble opinion you could spend your energy on a more important matter, since we have good ways (and better ones?) to run our CL apps. Unless you *really know* why you need it. My 2c!

**Reply by fukamachi** [reply]
I didn't know Hunchentoot supports FastCGI.
cl-fastcgi is finally back to the Quicklisp dist, so we can add fcgi handler again.

**Reply by daninus14** [reply]
> In my humble opinion you could spend your energy on a more important matter, since we have good ways (and better ones?) to run our CL apps. Unless you _really know_ why you need it. My 2c!

Thanks. Good point. Just for efficiency but to be honest it's not a bottleneck right now. You are right.

**Comment by vindarel** [comment]
See https://lispcookbook.github.io/cl-cookbook/web.html#running-behind-nginx for pointers on how to set-up a reverse proxy. It's a good solution.
